### PR TITLE
Add Index Stats API

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Add Index Stats API

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
@kolchfa-aws - Is there an issue associated with this or is this the one that is tied to a 2.10 feature?

**Comment by kolchfa-aws** [comment]
It's one of the missing APIs.

**Review by jed326 — APPROVED** [review]

**Review by Naarcha-AWS — APPROVED** [review]

File: _api-reference/index-apis/stats.md [hunk 1]
```diff
@@ -0,0 +1,647 @@
+---
+layout: default
+title: Stats
+parent: Index APIs
+nav_order: 72
+---
+
+# Index stats 
+
+The Index Stats API provides index statistics. For data streams, the API provides statistics for the stream's backing indexes. By default, the returned statistics are index-level. To receive shard-level statistics, set the `level` parameter to `shards`.
```
Comment (natebower) [inline _api-reference/index-apis/stats.md:10]:
```suggestion
The Index Stats API provides index statistics. For data streams, the API provides statistics for the stream's backing indexes. By default, the returned statistics are index level. To receive shard-level statistics, set the `level` parameter to `shards`.
```

File: _api-reference/index-apis/stats.md [hunk 2]
```diff
@@ -0,0 +1,647 @@
+---
+layout: default
+title: Stats
+parent: Index APIs
+nav_order: 72
+---
+
+# Index stats
```
Comment (natebower) [inline _api-reference/index-apis/stats.md:8]: Should "Stats" be capitalized here?

File: _api-reference/index-apis/stats.md [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+
+### Metrics
+
+The following table lists all available metric groups.
+
+Metric | Description
+:--- |:----
+`_all` | Return all statistics. 
+`completion` | Completion suggester statistics. 
+`docs` | Returns the number of documents and the number of deleted docs that have not yet been merged. Index refresh operations can affect this statistic.
```
Comment (natebower) [inline _api-reference/index-apis/stats.md:40]:
```suggestion
`docs` | Returns the number of documents and the number of deleted documents that have not yet been merged. Index refresh operations can affect this statistic. 
```

Comment (natebower) [inline _api-reference/index-apis/stats.md:40]: "Returns the total number of documents and the number of deleted documents..."?

File: _api-reference/index-apis/stats.md [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+`fielddata` | Field data statistics. 
+`flush` | Flush statistics. 
+`get` | Get statistics, including missing stats. 
+`indexing` | Indexing statistics. 
+`merge` | Merge statistics. 
+`query_cache` | Query cache statistics. 
+`refresh` | Refresh statistics. 
+`request_cache` | Shard request cache statistics. 
+`search` | Search statistics, including suggest operation statistics. Search operations can be associated with one or more groups. You can include statistics for custom groups by providing a `groups` parameter, which accepts a comma-separated list of group names. To return statistics for all groups, use `_all`. 
+`segments` | Statistics about memory use of all open segments. If the `include_segment_file_sizes parameter` is `true`, this metric includes the aggregated disk usage of each Lucene index file.
```
Comment (natebower) [inline _api-reference/index-apis/stats.md:50]: Is "parameter" supposed to be inside the code font?

File: _api-reference/index-apis/stats.md [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+`warmer` | Warmer statistics. 
+
+## Query parameters
+
+The following table lists the available query parameters. All query parameters are optional.
+
+Parameter | Data type | Description 
+:--- | :--- | :--- 
+`expand_wildcards` | String | Specifies the type of indexes to which wildcard expressions can expand. Supports comma-separated values. Valid values are: <br> - `all`: Expand to all open and closed indexes, including hidden indexes <br> `open`: Expand to open indexes <br> - `closed`: Expand to closed indexes <br> - `hidden`: Include hidden indexes when expanding. Must be combined with `open`, `closed`, or both <br> - `none`: Do not accept wildcard expressions <br> Default is `open`.
+`fields` | String | A comma-separated list or a wildcard expression specifying fields to include in the statistics. Specifies the default field list if neither `completion_fields` nor `fielddata_fields` are provided.
```
Comment (natebower) [inline _api-reference/index-apis/stats.md:62]:
```suggestion
`fields` | String | A comma-separated list or a wildcard expression specifying fields to include in the statistics. Specifies the default field list if neither `completion_fields` nor `fielddata_fields` is provided.
```

File: _api-reference/index-apis/stats.md [hunk 6]
```diff
@@ -0,0 +1,647 @@
...
+`completion_fields` | String | A comma-separated list or wildcard expression specifying fields to include in field-level `completion` statistics.
+`fielddata_fields` | String | A comma-separated list or wildcard expression specifying fields to include in field-level `fielddata` statistics.
+`forbid_closed_indices` | Boolean | Specifies not to collect statistics for closed indexes. Default is `true`.
+`groups` | String | A comma-separated list of search groups to include in the `search` statistics.
+`level` | String | Specifies the level to aggregate statistics. Valid values are: <br> - `cluster`: Cluster-level statistics, `indices`: Index-level statistics <br> - `shards`: Shard-level statistics. <br> Default is `indices`.
```
Comment (natebower) [inline _api-reference/index-apis/stats.md:67]: "the level used to aggregate"? Also, should `indices` be its own bullet point? And they should all end with a period.

File: _api-reference/index-apis/stats.md [hunk 7]
```diff
@@ -0,0 +1,647 @@
...
+`translog` | Translog statistics. 
+`warmer` | Warmer statistics. 
+
+## Query parameters
+
+The following table lists the available query parameters. All query parameters are optional.
+
+Parameter | Data type | Description 
+:--- | :--- | :--- 
+`expand_wildcards` | String | Specifies the type of indexes to which wildcard expressions can expand. Supports comma-separated values. Valid values are: <br> - `all`: Expand to all open and closed indexes, including hidden indexes <br> `open`: Expand to open indexes <br> - `closed`: Expand to closed indexes <br> - `hidden`: Include hidden indexes when expanding. Must be combined with `open`, `closed`, or both <br> - `none`: Do not accept wildcard expressions <br> Default is `open`.
```
Comment (natebower) [inline _api-reference/index-apis/stats.md:61]: Let's put periods at the end of each L2 bullet.

**Review by natebower — COMMENTED** [review]
@kolchfa-aws Please see my comments and changes and let me know if you have any questions. Thanks!

File: _api-reference/index-apis/stats.md [hunk 8]
```diff
@@ -0,0 +1,647 @@
...
+```json
+GET /testindex/_stats
+```
+{% include copy-curl.html %}
+
+#### Example response
+
+By default, the returned statistics are aggregated in the `primaries` and `total` aggregations. The `primaries` aggregation contains statistics for the primary shards. The `total` aggregation contains statistics for both primary and replica shards.
+
+<details closed markdown="block">
```
[the remark this answers is not in the document]
Comment (kolchfa-aws) [reply _api-reference/index-apis/stats.md:82]: In our template, we don't have an introductory sentence. I'll add one here though because it has more info.

**Pull request merged**